packages/tests/src/lib/index.ts

```diff
@@ -216,34 +238,106 @@ class MultipleNodesMessageCollector {
     const startTime = Date.now();
     const pubsubTopic = options?.pubsubTopic || DefaultTestPubsubTopic;
     const timeoutDuration = options?.timeoutDuration || 400;
+    const maxTimeout = Math.min(timeoutDuration * numMessages, 30000);
+    const exact = options?.exact || false;
+
+    try {
+      while (Date.now() - startTime < maxTimeout) {
```

Comment:
is loop needed here? we can hit endless cycle

Reply:
valid concern; moved to promsies + timeout: e095c4799dc3391e9a9dd2deecc34f42e736a54c